### ISSUE TITLE
Multiply blocks and notes instantly with ''Paint''.

### ISSUE THREAD

**Issue by Gabrielxd195** [LMMS/lmms#4694]
I was observing that fl studio has a tool called '' paint tool '', which is used to multiply blocks. the icon has a brush shape.

![kazam_screenshot_00013](https://user-images.githubusercontent.com/26127693/47884941-1e58ad00-de09-11e8-8d65-c202bf847ead.png)

Although this tool in fl studio is a bit different, in LMMS the '' paint '' tool would work both in the piano roll editor and in the song editor, and in the latter basically it would work like this:

1- In the song editor, you right click on a block of any editor.
2- In the menu you give '' copy ''.
3- In the tool bar select the tool ''paint''.
4- And with the selected tool, you leave the left click and drag it over the desired space in the song editor, while doing this, the blocks are automatically and instantly stuck side by side without overlapping.
5- And in this way we have copied or multiplied a block instantly without using the traditional copy and paste or the '' ctrl '' + drag.

Basically the tool would work as a '' paste '' but multiplying and distributing the copies over the spaces where the mouse pointer passes (similar to what happens in fl studio).

Within the piano roll, the function would be different.

1- We select a type of measure of notes (for example: round, white, etc).
2- Select the tool '' paint ''.
3- We left the right click and move the mouse pointer where we want. and ready. This way we avoid being constantly clicking. I do not know what you think of the idea, but I've seen that fl studio users use it more than "copy and paste". Greetings.

**Comment by Spekular** [comment]
Since this has been brought up in a new issue, I have to say that I think this is an inelegant solution to wanting to repeat a clip, at least for MIDI and automation. Defining a window on the clip content + a start and end position for the clip and letting the windowed region repeat for the clip duration instead could replace this painting with a resize, meaning there's no need to switch mode. Add a knife tool to all clips and you could split out + make unique sections as necessary.

**Comment by allejok96** [comment]
I agree. Make midi clips work like BB clips and just drag the right edge to extend them and make them repeat (and paint the repeated part in a different shade)
